src/par_schedule.rs

```diff
@@ -0,0 +1,187 @@
+extern crate priority_queue;
+use priority_queue::PriorityQueue;
+use crate::priority::Priority;
+use crate::agent::Agent;
+use crate::agentimpl::AgentImpl;
+use crate::state::State;
+use std::sync::Mutex;
+
+static THREAD_NUM:usize = 4;
```

Comment:
dynamically compute the number of thread, and also add some option to set the number of threads

Reply:
add some option to have the schedule of agents parallel or not.
You have also to consider the different versions of the field. When the number of threads is equal to 1, you must not use double buffering.